src/components/GifCard/GifCard.js

```diff
@@ -0,0 +1,100 @@
+import React, {Component} from 'react';
+import {bindActionCreators} from "redux";
+import { connect } from 'react-redux';
+import {
+    GifContainer,
+    Gif,
+    GifOverlay,
+    GifTitle,
+    GifOverlayLoading,
+    FavoriteBtn
+} from './GifCard.styled';
+import favoriteGifsActions from "../../actions/favoriteGifs";
+
+const LOCAL_STORAGE_FAV_GIFS = 'favoriteGifs';
+
+class GifCard extends Component {
+
+    constructor(props) {
+        super(props);
+
+        let isFavorite = this.gifInLocalStorage();
```

Comment:
Remover that the only source of truth should be the Redux store.

Reply:
I have a doubt on that. As well, I though to use Redux to deicide if the gifven Gif is Favorite, but I didn't know how to, because... I would have to define as many `isFavorite` state per gif, I mean, something like this:

```
{
    gifId1: (True/False),
    gifId2: (True/False),
    gifId3: (True/False)
}
```

And it will get bigger depending on the selected favorite gifs.

What would you recommend me??